### PR TITLE
fix: default ServiceContainer to persistent LanceDB storage

### LINKED ISSUE #39: Default StorageConfig should persist to LanceDB with sensible URI
## Problem

`StorageConfig()` defaults to `uri="./archetype_data"` and `backend=LANCEDB`, but when used through `ServiceContainer` the world doesn't always persist because the factory path may use in-memory querier/updater.

## Desired behavior

- Default `StorageConfig()` should always persist to LanceDB at `./archetype_data`
- If the URI path is unwritable, raise a clear error
- Every world created through `ServiceContainer` should be durable out of the box

## Acceptance criteria

- [ ] `ServiceContainer` → `create_world(config)` without explicit `StorageConfig` still persists
- [ ] Data survives process restart (verified with a test)
- [ ] Existing tests still pass

@claude implement this

### PR DESCRIPTION
**Pull request by everettVT**
## Summary
- **#39**: Default `ServiceContainer` to durable LanceDB storage with sensible URI
- **#115**: Fix OpenAPI doc generator edge cases (anyOf unwrap, param type corrections)

## Test plan
- [x] CI passes (`make ci`)
- [ ] Verify default storage persists across restarts

Closes #39

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/app/test_default_storage_persistence.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            await container.world_service.create_world(WorldConfig(name="explicit"), explicit)
+            # The multiton key on StorageService encodes uri+namespace.
+            keys = list(container.storage_service._instances.keys())
+            assert any("custom" in k and "explicit_ns" in k for k in keys), keys
+        finally:
+            await container.shutdown()
+
+    @pytest.mark.asyncio
+    async def test_unwritable_uri_raises_clear_error(self, tmp_path):
+        """Unwritable URIs must raise PermissionError with a helpful message."""
```
[the remark this answers is not in the document]
Comment (everettVT) [reply tests/app/test_default_storage_persistence.py:108]: Superseded on current main: tests/app/test_default_storage_persistence.py is absent at 8329d506da661c22fb7c4f31fa4e13b783a531f3, so the reviewed implementation no longer exists to amend. Any equivalent behavior in the replacement architecture remains part of the current-main audit.

File: tests/app/test_default_storage_persistence.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+        container = ServiceContainer()
+        try:
+            world = await container.world_service.create_world(WorldConfig(name="default"))
+            # The querier/updater must be backed by the real LanceDB store,
+            # not an in-memory stub.
+            store = container.storage_service._instances
+            assert store, "StorageService should have at least one backend instance"
+            any_store = next(iter(store.values()))[0]
+            assert isinstance(any_store, AsyncLancedbStore)
```
[the remark this answers is not in the document]
Comment (everettVT) [reply tests/app/test_default_storage_persistence.py:43]: Superseded on current main: tests/app/test_default_storage_persistence.py is absent at 8329d506da661c22fb7c4f31fa4e13b783a531f3, so the reviewed implementation no longer exists to amend. Any equivalent behavior in the replacement architecture remains part of the current-main audit.

File: tests/app/test_default_storage_persistence.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    async def test_explicit_storage_config_is_honored(self, tmp_path):
+        """Passing a StorageConfig explicitly must not be overridden by defaults."""
+        explicit = StorageConfig(uri=str(tmp_path / "custom"), namespace="explicit_ns")
+
+        container = ServiceContainer()
+        try:
+            await container.world_service.create_world(WorldConfig(name="explicit"), explicit)
+            # The multiton key on StorageService encodes uri+namespace.
+            keys = list(container.storage_service._instances.keys())
+            assert any("custom" in k and "explicit_ns" in k for k in keys), keys
```
[the remark this answers is not in the document]
Comment (everettVT) [reply tests/app/test_default_storage_persistence.py:102]: Superseded on current main: tests/app/test_default_storage_persistence.py is absent at 8329d506da661c22fb7c4f31fa4e13b783a531f3, so the reviewed implementation no longer exists to amend. Any equivalent behavior in the replacement architecture remains part of the current-main audit.
